server/chat-plugins/chatlog.ts

```diff
@@ -731,14 +731,13 @@ export const commands: ChatCommands = {
 	sl: 'searchlogs',
 	searchlog: 'searchlogs',
 	searchlogs(target, room) {
-		room = this.requireRoom();
 		target = target.trim();
 		const args = target.split(',').map(item => item.trim());
 		if (!target) return this.parse('/help searchlogs');
 		let date = 'all';
 		const searches: string[] = [];
 		let limit = '500';
-		let tarRoom = room.roomid;
+		let tarRoom = room?.roomid;
```

Comment:
This seems redundant with

```
		const curRoom = tarRoom ? Rooms.search(tarRoom) : room;
```

It feels like the entire thing should be refactored. Let me think.

Reply:
```ts
		for (const arg of args) {
			if (arg.startsWith('room:')) {
				const id = arg.slice(5);
				room = Rooms.search(id as RoomID);
				if (!room) {
					return this.errorReply(`Room "${id}" not found.`);
				}
			} else if (arg.startsWith('limit:')) {
				limit = arg.slice(6);
			} else if (arg.startsWith('date:')) {
				date = arg.slice(5);
			} else {
				searches.push(arg);
			}
		}
		if (!room) {
			return this.parse(`/help searchlogs`);
		}
```

No need for `curRoom` or `tarRoom` which don't really make much sense to me anyway.